client/src/components/ColorList.js

```diff
@@ -76,8 +115,36 @@ const ColorList = ({ colors, updateColors }) => {
           </div>
         </form>
       )}
-      <div className="spacer" />
-      {/* stretch - build another form here to add a color */}
+      {add && (
+        <form onSubmit={handleNewColor}>
+          <legend>add color</legend>
+          <label>
+            color name:
+            <input
+              onChange={e =>
+                setNewColor({ ...newColor, color: e.target.value })
+              }
+              value={newColor.color}
```

Comment:
Because the onChange directly contains the function to update state, we have no need for a `name` property 👍